### DISCUSSION TITLE
请问如何使一个View展示loading状态？

### DISCUSSION THREAD

**Discussion by MilkWangStudio** [opensumi/core#3111] [Q&A] [answered]
通过切换按钮需要动态更新目录树，切换时希望这个FileExplorer展示loading状态，请问有什么办法吗？

<img width="351" alt="image" src="https://github.com/opensumi/core/assets/15866249/8ff447c6-d3dc-4b60-8d5e-92397621fdd9">

**Comment by erha19** [comment]
@MilkWangStudio 你这里是修改了文件树模块还是自己写了个新模块？文件树模块在根文件请求的时候应该是会 Loading 的，引入了 ProgressBar 组件。

```ts
import { Progress } from '@opensumi/ide-core-browser/lib/progress/progress-bar';
```

**Reply by MilkWangStudio** [reply]
修改了文件树模块，点击这个按钮后，我会把所FileSystem的文件清空，内存缓存的文件数据也清空，然后重新执行HttpFileService的初始化逻辑。

这个Progress我看到是在 FileTreeModelService.onFileTreeModelChange里监听控制的，有什么办法能触发么？

**Reply by erha19** [reply, marked as answer]
既然已经修改了文件树模块，那建议你可以自行对 Progress 这个组件做一下状态控制，目前这里没有手动触发的逻辑，整体是跟随文件树的业务逻辑编码的。

**Reply by MilkWangStudio** [reply]
好的，感谢回复，我自己修改下吧。
